### DISCUSSION TITLE
parseAbi type error

### DISCUSSION THREAD

**Discussion by Confucian-e** [wevm/viem#689] [Question]
How to solve this?
<img width="972" alt="image" src="https://github.com/wagmi-dev/viem/assets/46948182/611a00e0-1649-4356-9507-634cbd0c9987">

**Comment by Raiden1411** [comment]
Try declaring the string[] as const

**Reply by Confucian-e** [reply]
That worked, thx
```ts
const stake = ['function stake(address token_addr, uint256 _amount) external returns (bool success)']

export const abi = parseAbi(stake)
```

**Comment by tmm** [comment]
What version of TypeScript are you using? Can you through this into a [TypeScript Playground](https://www.typescriptlang.org/play) and create a reproduction?

**Reply by Confucian-e** [reply]
Well I checked my `package.json`, it shows `"typescript": "^4.4.2",`

**Reply by tmm** [reply]
got it! viem requires at least TypeScript 4.9.4. More info here: https://viem.sh/docs/typescript.html#typescript

**Reply by Confucian-e** [reply]
Got it! I will update my dependency, thx
